include/vara/Feature/FeatureModel.h

```diff
@@ -34,12 +34,12 @@ class FeatureModel {
   using OrderedFeatureTy = OrderedFeatureVector;
   using ConstraintTy = Constraint;
   using ConstraintContainerTy = std::vector<std::unique_ptr<ConstraintTy>>;
-  using RelationshipTy = Relationship;
-  using RelationshipContainerTy = std::vector<std::unique_ptr<RelationshipTy>>;
+  using RelationshipContainerTy = std::vector<std::unique_ptr<Relationship>>;
 
+  FeatureModel() = default;
```

Comment:
This is kind of required, if we want to build a FM only using transactions, so we start with an empty tree.
It may be reasonable to initialize a root feature at this point, which would make that explicit step in the builder obsolete.
Currently completely empty FM (without any nodes and especially without a root) are valid.

Reply:
I think making the FMbuilder a friend and this private may be a better approach.

We might want to support an constructor without features, constraints etc though.